### ISSUE TITLE
OpenComputers AE2 API

### ISSUE THREAD

**Issue by ghost** [MightyPirates/OpenComputers#3279]
I want to use OpenComputers with AE2 to auto craft Botania runes. I looked at the OpenComputers page for AE2 but i couldn't figure out how to get a specific craftable item. I know you can filter it but what's the filter table format? Also what's the format of `getCraftables()` functions return table format? What does returned table contain and what's the keys for it?

By format i meant what to write in tables.

Thanks for the help.

(Sorry to write here couldn't find anywhere where i could get answer)

**Comment by brisingraerowing** [comment]
It would be better to ask on [the forums](https://oc.cil.li/). There are also Discord and IRC channels (which are linked together).

**Comment by payonel** [comment]
the table from `getAllCraftables` is a lua array [1, n] of the craftable objects in the network
if you wanted to craft recipe 1 in the list, you could do this:

```lua
  local m = component.me_controller
  local c = m.getAllCraftables()[1]
  local status = c.request(1) -- craft 1
  while not status.isDone() do
    if status.isCanceled() then
      print("the craft was canceled")
      break
    end
    os.sleep(1)
  end
  print("craft job wait exit")
```

a few things to note:
1. our ae2 support is discontinued in 1.7.10 and 1.10 - any further fixes/additions will only go to 1.12
2. i just pushed significant fixes to our crafting support. this will be available in our next release (after 1.7.5) or you can get it now in our dev builds: https://ci.cil.li/job/OpenComputers-MC1.12/
3. we are using the isDone and isCanceled api directly to AE, to their meaning is important. isDone is false when the job is canceled. It is only true when the job reaches successful completion.
